webapp/src/api.rs

```diff
@@ -17,3 +26,45 @@ pub async fn version() -> Json<Version> {
 pub async fn get_unused_address() -> impl IntoResponse {
     api::get_unused_address().0
 }
+
+#[derive(Serialize)]
+pub struct Balance {
+    on_chain: u64,
+    off_chain: u64,
+}
+
+pub async fn get_balance(
+    State(subscribers): State<Arc<AppSubscribers>>,
+) -> Result<Json<Balance>, AppError> {
+    ln_dlc::refresh_wallet_info().await?;
+    let balance = subscribers
+        .wallet_info()
+        .map(|wallet_info| Balance {
+            on_chain: wallet_info.balances.on_chain,
+            off_chain: wallet_info.balances.off_chain,
+        })
+        .unwrap_or(Balance {
+            on_chain: 0,
+            off_chain: 0,
+        });
```

Comment:
We should probably return an error to the caller instead of setting it to 0 here. Could be quite scary if the balance shows suddenly 0.

Reply:
Oh, you did it already below. nice

Reply:
I suggest we do that when we consider implementing polling or push. Otherwise, the user would simply get an error only because his request came in earlier than the wallet info event has been pusblished.